packages/protocol/contracts-0.8/common/EpochManager.sol

```diff
@@ -319,4 +336,42 @@ contract EpochManager is
       CELOequivalent
     );
   }
+
+  /**
+   * @notice Sends the allocated epoch payment to a validator, their group, and
+   *   delegation beneficiary.
+   * @param validator Account of the validator.
+   */
+  function sendValidatorPayment(address validator) external {
+    IAccounts accounts = IAccounts(getAccounts());
+    address signer = accounts.getValidatorSigner(validator);
+
+    FixidityLib.Fraction memory totalPayment = FixidityLib.newFixed(
+      validatorPendingPayments[signer]
+    );
+
+    IValidators validators = getValidators();
+    address group = validators.getValidatorsGroup(validator);
+    (, uint256 commissionUnwrapped, , , , , ) = validators.getValidatorGroup(group);
+
+    uint256 groupPayment = totalPayment.multiply(FixidityLib.wrap(commissionUnwrapped)).fromFixed();
+    FixidityLib.Fraction memory remainingPayment = FixidityLib.newFixed(
+      totalPayment.fromFixed() - groupPayment
```

Comment:
just checking... `groupPayment` here is a Fixidity? in that case `sub()` is same as `-`?

Reply:
Yes, both values are *unwrapped* Fixidity. In general, this logic is copied over from `Validators._distributeEpochPaymentsFromSigner`. The `.sub()` there is actually the SafeMath subtraction on regular `uint256`s, not the Fixidity `.subtract()`. So `-` is the correct 0.8 equivalent.

Reply:
Is there any reason why we don't do the subtraction in fixidity for better precision and only then convert to uint256 ?

Reply:
We still use the same precision, as this is on unwrapped Fixidity values. I think this may have been a gas savings measure?